torch/nn/quantized/modules/conv.py

```diff
@@ -352,11 +381,9 @@ def forward(self, input):
         # https://github.com/pytorch/pytorch/issues/23890
         if len(input.shape) != 3:
             raise ValueError("Input shape must be `(N, C, L)`!")
-        if self.padding_mode != 'zeros':
-            # Padding in Conv1d is stored as (p, p), need to get (p,)
-            _reversed_padding_repeated_twice = _reverse_repeat_padding(self.padding[:1])
-            input = F.pad(input, _reversed_padding_repeated_twice,
-                          mode=self.padding_mode)
+        if isinstance(self.padding, str) or self.padding_mode != 'zeros':
```

Comment:
in the floating point conv the corresponding check is `if self.padding_mode != 'zeros':`, is there a reason it is different here?

Reply:
Yes, the FP implementation for the convolution supports different types for convolution. However, in quantization conv (or more accurately in the prepacking cpp code), we don't support taking strings as padding. Hence, we do it here.

Reply:
One clarification: the FP convolution supports ONLY zero padding (afaik). Basically, that means when calling FP convolution with "reflect" or "circular" padding, it calls the `F.pad` before calling the `F.conv`.

Also, the reason for this additional check is that the FP conv can take string arguments, while quantized conv kernel cannot: internally, the FP conv computes the amount of padding that is required, if the padding is `same`, and only after that calls the conv kernel. In our case, we compute the amount of padding outside the Cpp kernel, and pre-compute it. After that we call F.pad to be able to support "asymmetric" padding (which by design is not supported in conv).